tools/syz-declextract/syz-declextract.cpp

```diff
@@ -126,13 +126,24 @@ class SyscallMatcher : public MatchFinder::MatchCallback {
 
 class NetlinkPolicyMatcher : public MatchFinder::MatchCallback {
 private:
-  const std::string nlaToSyz(const Expr *const policyType) {
-    // NOTE:This check is for when the policy is missing the field `type`.
-    // TODO:Gather information from other defined fields to better specify a type.
-    if (!policyType->getEnumConstantDecl()) {
-      return "intptr";
+  std::unordered_map<std::string, uint8_t> nlaEnumToU8;
+  std::unordered_map<uint8_t, std::string> u8ToNlaEnum;
+  void nlaEnum(const MatchFinder::MatchResult &Result) {
+    const auto &num = Result.Nodes.getNodeAs<EnumDecl>("NLA_ENUM");
```

Comment:
Please add comments here as to what's in NLA_ENUM and what's stored u8ToNlaEnum in the end. Some examples would help.
I have no glue as to what's happening here.

Reply:
NLA_ENUM refers to https://elixir.bootlin.com/linux/v6.10/source/include/net/netlink.h#L172
We store the Enum values to string conversion in this map.
This is used later when assessing the type as an expression instead of a straight-up Enum, which is needed when the type is assigned via a macro that expands to an expression e.g. https://elixir.bootlin.com/linux/v6.10/source/drivers/net/wireguard/netlink.c#L25

Reply:
For example nlaEnumToU8["NLA_UNSPEC"] would result in zero. But u8ToNlaEnum is never used. I'll remove it.

Reply:
Done!